subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/msvcpp/DefaultWindowsKitWindowsSdkLocator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.toolchain.internal.msvcpp;
+
+import net.rubygrapefruit.platform.WindowsRegistry;
+import org.gradle.util.VersionNumber;
+
+import java.io.File;
+
+public class DefaultWindowsKitWindowsSdkLocator extends AbstractWindowsKitLocator<WindowsKitWindowsSdk> implements WindowsKitWindowsSdkLocator {
+    private static final String COMPONENT_NAME = "um";
+    private static final String DISPLAY_NAME = "Windows SDK";
+    private static final String RC_EXE = "rc.exe";
+
+    public DefaultWindowsKitWindowsSdkLocator(WindowsRegistry windowsRegistry) {
+        super(windowsRegistry);
+    }
+
+    @Override
+    String getComponentName() {
+        return COMPONENT_NAME;
+    }
+
+    @Override
+    String getDisplayName() {
+        return DISPLAY_NAME;
+    }
+
+    @Override
+    boolean isValidComponentBaseDir(File baseDir) {
+        return new File(baseDir, "bin/x86/" + RC_EXE).exists();
+    }
+
+    @Override
+    boolean isValidComponentIncludeDir(File includeDir) {
+        return new File(includeDir, "windows.h").exists();
+    }
+
+    @Override
+    boolean isValidComponentLibDir(File libDir) {
+        return new File(libDir, "x86/kernel32.lib").exists();
```

Comment:
Same as above. I know this is what we were doing previously, but may be worth checking and tracking via an issue if it's a limitation.

Reply:
I added validation for both x86 and x64 libraries as well as for `rc.exe`.  The interesting thing is that for the arm platform, there is no `rc.exe`.  This holds true for the Windows 8 and 8.1 kits as well (although we do return a path to the "arm/bin" location if requested).  I'm not aware of any defect reports on this, so I'm wondering if this is something that just isn't done (i.e. using `rc.exe` for an arm platform).  We can certainly change it to return say "x86/bin", but I'm not sure if this would be any more correct.  I just don't know enough about it and some googling did not turn anything up.  As it is, the Windows 10 sdk will do the same thing as the older sdks - return a path to `bin/arm/rc.exe` if requested.  Do you know anything about this?